From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: GameCubeGBA <95232096+GameCubeGBA@users.noreply.github.com>
Date: Wed, 15 Dec 2021 15:26:06 -0500
Subject: [PATCH] 3760: Fix typo causing "hasRangeMatch" to always be true

It should be set to false initially until a match is found. Because of this error, checkInsideThunkFunction will always think there is a range match, which could cause disastrous results.
---
 .../java/ghidra/app/util/bin/format/pdb/ApplyFunctions.java     | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Ghidra/Features/PDB/src/main/java/ghidra/app/util/bin/format/pdb/ApplyFunctions.java b/Ghidra/Features/PDB/src/main/java/ghidra/app/util/bin/format/pdb/ApplyFunctions.java
index 5a4f3975b..1573e3617 100644
--- a/Ghidra/Features/PDB/src/main/java/ghidra/app/util/bin/format/pdb/ApplyFunctions.java
+++ b/Ghidra/Features/PDB/src/main/java/ghidra/app/util/bin/format/pdb/ApplyFunctions.java
@@ -112,7 +112,7 @@ class ApplyFunctions {
 			return null;
 		}
 		AddressSet thunkBody = new AddressSet(thunkFunction.getBody());
-		boolean hasRangeMatch = true;
+		boolean hasRangeMatch = false;
 		AddressRangeIterator ari = thunkBody.getAddressRanges(true);
 		while (ari.hasNext()) {
 			if (monitor.isCancelled()) {
-- 
2.38.1

